src/ws/cockpitauth.c

```diff
@@ -780,6 +780,21 @@ on_transport_control (CockpitTransport *transport,
     }
   else if (g_str_equal (command, "authorize"))
     {
+      const gchar *challenge;
+
+      /* handle x-host-key challenge */
+      if (cockpit_json_get_string (options, "challenge", NULL, &challenge) && g_strcmp0 (challenge, "x-host-key") == 0)
+        {
+          const gchar *cookie;
+          g_return_val_if_fail (cockpit_json_get_string (options, "cookie", NULL, &cookie), FALSE);
+
+          /* return a negative answer; we handle unknown hosts interactively, or want to fail on them */
+          g_debug ("received x-host-key authorize challenge");
+          send_authorize_reply (session->transport, cookie, "");
```

Comment:
According to cockpitsshrelay.c we should be setting "* invalid key" if we want it to fail. Is this correct? Either way this is begging for documentation at this point. Can we update doc/authorize.md?

Reply:
So `""` is the legacy one and `"* invalid key"` the current one ([code](https://github.com/cockpit-project/cockpit/blob/master/src/ssh/cockpitsshrelay.c#L318)). Indeed I noticed that with values `""` and `"*"` (which I also tried), what actually arrives in cockpit-ssh is always `* invalid key`. So I'll change it to that and document this, too.

Reply:
@stefwalter: Actually, [challenge_for_auth_data()](https://github.com/cockpit-project/cockpit/blob/master/src/ssh/cockpitsshrelay.c#L299) only treats `""` as invalid . With `"* invalid key"`, challenge_for_knownhosts_data() returns with a value and triggers the

    authorize challenge reported key for 127.0.0.1:43183 which is not known to cockpit_is_host_known()

warning. I can silence this, but this would mean more special cases. What's wrong with using an empty response?

Reply:
Lets use empty then, but:

- Use it consistently
- Document it.

Reply:
I [documented it](https://github.com/martinpitt/cockpit/blob/drop-challenge-preconnect-hack/doc/authentication.md) now (separate commit), and also changed the unit tests to use an `INVALID_KEY` macro, so that it's both more obvious what it is, and we can also change it centrally if we want to.